src/test/java/com/nimbusds/jose/JWEObjectJSONTest.java

```diff
@@ -0,0 +1,114 @@
+package com.nimbusds.jose;
+
+import com.nimbusds.jose.crypto.*;
+import com.nimbusds.jose.jwk.Curve;
+import com.nimbusds.jose.jwk.ECKey;
+import com.nimbusds.jose.jwk.OctetKeyPair;
+import com.nimbusds.jose.jwk.gen.ECKeyGenerator;
+import com.nimbusds.jose.jwk.gen.OctetKeyPairGenerator;
+import com.nimbusds.jose.util.Base64URL;
+import junit.framework.TestCase;
+
+public class JWEObjectJSONTest extends TestCase {
+
+    private static ECKey generateECJWK(final Curve curve, String kid)
+            throws Exception {
+        return new ECKeyGenerator(curve).keyID(kid).generate();
+    }
+
+    private static OctetKeyPair generateOckeJWK(final Curve curve, String kid)
+            throws Exception {
+        return new OctetKeyPairGenerator(curve).keyID(kid).generate();
+    }
+
+    public void testECDH_1PU_multi_encrypt_decrypt() throws Exception {
+        JWEHeader header = new JWEHeader.Builder(JWEAlgorithm.ECDH_1PU_A256KW, EncryptionMethod.A256CBC_HS512)
+                .agreementPartyVInfo(Base64URL.encode("Alice"))
+                .agreementPartyUInfo(Base64URL.encode("Bob"))
+                .build();
+
+        ECKey aliceKey = generateECJWK(Curve.P_521, "alice");
+        ECKey bobKey = generateECJWK(Curve.P_521, "bob");
+        ECKey charlieKey = generateECJWK(Curve.P_521, "charlie");
+
+        JWEObjectJSON jwe = new JWEObjectJSON(header, new Payload("Hello, world"));
+        ECDH1PUEncrypterMulti encrypterMulti = new ECDH1PUEncrypterMulti(aliceKey, new ECKey[]{ bobKey, charlieKey });
+        jwe.encrypt(encrypterMulti);
+
+        String json = jwe.serialize();
+
+        JWEObjectJSON decrypted = JWEObjectJSON.parse(json);
+        ECDH1PUDecrypterMulti decryptor = new ECDH1PUDecrypterMulti(aliceKey, new ECKey[]{ bobKey, charlieKey });
```

Comment:
What if
- a recipient key not present in JSON is passed (so there is no corresponding `kid`), so we passed more keys than needed
- a recipient key present in JSON and required for decryption is not passed, so we passed less keys than needed.
Will it raise an exception? Are we planning to add tests like this?